activesupport/test/core_ext/time_ext_test.rb

```diff
@@ -430,6 +430,10 @@ def test_change
     assert_equal Time.local(2005, 1, 2, 11, 22, 33, 8), Time.local(2005, 1, 2, 11, 22, 33, 2).change(nsec: 8000)
     assert_raise(ArgumentError) { Time.local(2005, 1, 2, 11, 22, 33, 8).change(usec: 1, nsec: 1) }
     assert_nothing_raised { Time.new(2015, 5, 9, 10, 00, 00, "+03:00").change(nsec: 999999999) }
+
+    with_env_tz "US/Eastern" do
+      assert_equal 0, Time.new(2005, 10, 30, 0, 59, 59).advance(hours: 1).change({}) - Time.new(2005, 10, 30, 0, 59, 59).advance(hours: 1)
```

Comment:
Is there a reason to use `assert_equal 0` instead of comparing the times directly?

Also, if we fix the root cause, we should test non-empty options:

```suggestion
      assert_equal Time.new(2005, 10, 30, 0, 59, 59).advance(hours: 1), Time.new(2005, 10, 30, 0, 59, 59).advance(hours: 1).change(year: 2005)
      assert_equal Time.new(2005, 10, 30, 2, 00, 01).advance(hours: -1), Time.new(2005, 10, 30, 2, 00, 01).advance(hours: -1).change(year: 2005)
```